concordium-std/src/errors.rs

```diff
@@ -0,0 +1,9 @@
+//! This internal module provides the error types used in the crate.
+
+#[derive(Debug)]
+pub enum ContractStateError {
```

Comment:
We already have a number of error types in `types.rs`, so I think this should live there as well 😊 
If you think that the errors should live in a separate error module, then we can move all of them in a separate PR at some point.

Reply:
I agree with Kasper. Do not partially move errors here. This belongs in types.rs.